internal/heartbeat/heartbeat.go

```diff
@@ -295,11 +315,14 @@ func (s *Heartbeat) initTicker(periodSeconds int64) {
 
 func (s *Heartbeat) Deregister() error {
 	s.log.Infof("stopping heartbeat ticker. DeviceID: %s", s.data.workloadManager.GetDeviceID())
-	if s.ticker != nil {
+	s.stopTicker()
+	return nil
+}
+
+func (s *Heartbeat) stopTicker() {
```

Comment:
Well not really; `HasStarted` only checks if the ticker is created and `stopTicker` is stoping the ticker so it needs to check that `ticker != nil` before stoping it, that's why I used `HasStarted` here. But I agree I could simply check for the `nil` as it's the only thing the matters here.
`HasStarted` could become more complex if we want to check other things other than the `nil` value so I'll keep the function.
And, for readability, it's better to separate `stopTicker` from `Deregister`, as again, deregister could become more complex than just stoping the ticker. This way it will be easier to make new changes to it